Source/WTF/wtf/URLParser.cpp

```diff
@@ -348,7 +220,9 @@ ALWAYS_INLINE bool URLParser::isForbiddenDomainCodePoint(CharacterType character
 
 ALWAYS_INLINE static bool shouldPercentEncodeQueryByte(uint8_t byte, const bool& urlIsSpecial)
 {
-    if (characterClassTable[byte] & QueryPercent)
+    if (byte > 0x7F)
```

Comment:
This is an unnecessary branch in very hot code.  I think it would be better to keep the entire table.